src/pieces/api/assets.py

```diff
@@ -0,0 +1,197 @@
+## ASSET CALLS 
+from .config import *
+from pieces.gui import show_error
+from pydantic import ValidationError
+from pieces_os_client.models.classification import Classification
+from pieces_os_client.rest import ApiException
+def create_new_asset(application, raw_string, metadata=None):
+    assets_api = pos_client.AssetsApi(api_client)
+
+    # Construct a Seed
+    seed = pos_client.Seed(
+        asset=pos_client.SeededAsset(
+            application=application,
+            format=pos_client.SeededFormat(
+                fragment=pos_client.SeededFragment(
+                    string=pos_client.TransferableString(raw=raw_string)
+                )
+            ),
+            metadata=metadata  # This should be constructed as per the SDK's definition
+        ),
+        type="SEEDED_ASSET"
+    )
+    
+    # Creating the new asset using the assets API
+    try:
+        created_asset = assets_api.assets_create_new_asset(transferables=False, seed=seed)
+        return created_asset
+    except ApiException as e:
+        show_error("An exception occurred when calling AssetsApi->assets_create_new_asset:" , e)
+        return None
+
+def get_asset_ids(max=None, **kwargs):
+    assets_api = pos_client.AssetsApi(api_client)
+
+    try:
+        # Call the API to get assets identifiers
+        api_response = assets_api.assets_identifiers_snapshot()
+
+        # Extract data from the response
+        data = api_response.to_dict()  # Convert the response to a dictionary
```

Comment:
any reason why we need to convert it so a dictionary?, might be roll over from the refactor as well, feel free to add a TODO to use class properties instead of converting to a map

Reply:
I am not the one who edited this